tools/copy_bom/src/util.rs

```diff
@@ -370,6 +370,9 @@ pub fn extract_dependencies_from_output(
     }
     for line in stdout.lines() {
         let line = line.trim();
+        if line.contains(" => not found") {
+            continue;
```

Comment:
If we print the missing shared object file's name and return an error immediately, we fix the "silently fail" bug.

Reply:
Yes. Since we don't need `autodep_allow_fail`, we don't have to change the `ld.so --list` now. It will report the missing files in it's stderr. 
```
root@aa5bc5828445:/usr/share/rats-tls/samples/occlum_workspace_server# /lib64/ld-linux-x86-64.so.2 --list ../rats-tls-server
../rats-tls-server: error while loading shared libraries: librats_tls.so.0: cannot open shared object file: No such file or directory
```

All we need to do is change the logic here:
https://github.com/occlum/occlum/blob/af516757ca0bf7499afcdbf459d47e994a385269/tools/copy_bom/src/util.rs#L368-L370

I will create another pr to fix this.